File: Lecture18: Insertion Sort/sort01.py

def printArray( arr, n):
    for i in range(0,n):
        print(arr[i],end=" ")

    print("\n")


def sortOne( arr, n):

    left = 0;right = n-1

    while(left < right):

        while(arr[left] == 0 and left < right) :
            left = left + 1
        

        while(arr[right] == 1 and left < right):
            right = right -1
        

        # agar yha pohoch gye ho, iska matlab
        # arr[left] == 1 and arr[right] == 0
        if(left < right):
            swap(arr,left,right)
            left=left+1
            right=right-1
  

def swap(arr, x, y):
    arr[x], arr[y] = arr[y], arr[x]
    
if __name__ == "__main__":

    arr = [1, 1, 0, 0, 0, 0, 1, 0]

    sortOne(arr, len(arr))
    printArray(arr, len(arr))

File: Lecture010 Solving LeetCode Arrays/sort01.cpp
#include<iostream>
using namespace std;

void printArray(int arr[], int n) {
    for(int i=0; i<n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

void sortOne(int arr[], int n) {

    int left = 0, right = n-1;

    while(left < right) {

        while(arr[left] == 0 && left < right )  {
            left++;
        }

        while(arr[right]==1 && left < right){
            right--;
        }

        //agar yha pohoch gye ho, iska matlab
        //arr[left]==1 and arr[right]==0
        if(left<right)
        {
            swap(arr[left], arr[right]);
            left++;
            right--;
        }
    }

}

int main() {

    int arr[8] = {1,1,0,0,0,0,1,0};

    sortOne(arr, 8);
    printArray(arr, 8);


    return 0;
}